compiler/rustc_middle/src/mir/interpret/allocation/provenance_map.rs

```diff
@@ -148,6 +148,17 @@ impl<Prov: Provenance> ProvenanceMap<Prov> {
         true
     }
 
+    /// Gets the provenances of all bytes (including from pointers) in a range.
+    pub fn get_range(
+        &self,
+        cx: &impl HasDataLayout,
+        range: AllocRange,
+    ) -> impl Iterator<Item = (Size, Prov)> {
```

Comment:
There's no point in including the offset since one cannot even know whether that's a 1-byte provenance or a whole-ptr provenance. So either drop the `Size`, or add an `Option<u8>` carrying that information.